### ISSUE TITLE
BUG:   api: corr & corrwith, in some case , the result will be error

### ISSUE THREAD

**Issue by DarkTale** [pandas-dev/pandas#53633]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
df = pd.DataFrame(data={'a': [1.21, 1.21, 1.21], 'b': [1.21, 1.21, 1.21]})
print(df.corr())
```

### Issue Description

output:
    a   b
a NaN NaN
b NaN NaN

if the column items are float  type and same, the result  will be random between nan and 1.
the same is corrwith.

### Expected Behavior

output:
    a  b
a  1.0 1.0
b  1.0 1.0

### Installed Versions

<details>
1.1.5
</details>

**Comment by UmbertoFasci** [comment]
@DarkTale The dtype of the columns do not create this pattern. The expected value is NaN when the values are all the same.

In other words, with no difference in variance and standard deviation (0), the result will always result in a `NaN` value. Essentially, you are attempting to divide 0 by 0. 

Please let me know if this response was helpful to you.

**Comment by DarkTale** [comment]
yeah, if 0/0,it should always be nan.  
thanks a lot.  may be my test data's problem.result to cov & std is a very small value near zero,but not zero.  

thanks a lot!



---- Replied Message ----
| From | Umberto ***@***.***> |
| Date | 06/13/2023 10:28 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [pandas-dev/pandas] BUG: api: corr & corrwith, in some case , the result will be error (Issue #53633) |

@DarkTale The dtype of the columns do not create this pattern. The expected value is NaN when the values are all the same.

In other words, with no difference in variance and standard deviation (0), the result will always result in a NaN value. Essentially, you are attempting to divide 0 by 0.

Please let me know if this response was helpful to you.

**Comment by mroeschke** [comment]
Closing as a usage question
